app/src/main/java/org/mifos/mobile/ui/fragments/TransferProcessFragment.kt

```diff
@@ -49,42 +47,128 @@ class TransferProcessFragment : BaseFragment(), TransferProcessView {
+            tvPayFrom.text = payload?.fromAccountNumber.toString()
+            tvPayTo.text = payload?.toAccountNumber.toString()
+            tvDate.text = payload?.transferDate
+            tvRemark.text = payload?.transferDescription
+        }
 
         return binding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        binding.btnStartTransfer.setOnClickListener {
-            startTransfer()
-        }
-        binding.btnCancelTransfer.setOnClickListener {
-            cancelTransferProcess()
+        viewModel.transferUiState.observe(viewLifecycleOwner) {
+            when (it) {
+                is TransferUiState.Loading -> showProgress()
+                is TransferUiState.TransferSuccess -> {
+                    hideProgress()
+                    showTransferredSuccessfully()
+                }
+                is TransferUiState.Error -> {
+                    hideProgress()
+                    showError(MFErrorParser.errorMessage(it.errorMessage))
+                }
+            }
         }
-        binding.btnClose.setOnClickListener {
-            closeClicked()
+
+        with(binding) {
+            btnStartTransfer.setOnClickListener {
+                startTransfer()
+            }
+            btnCancelTransfer.setOnClickListener {
+                cancelTransferProcess()
+            }
+            btnClose.setOnClickListener {
+                closeClicked()
+            }
         }
     }
 
     /**
      * Initiates a transfer depending upon `transferType`
      */
-    fun startTransfer() {
+    private fun startTransfer() {
         if (!Network.isConnected(activity)) {
             Toaster.show(binding.root, getString(R.string.internet_not_connected))
             return
         }
+        val fromOfficeId = fromAccountOption?.officeId
+        val fromClientId = fromAccountOption?.clientId
+        val fromAccountType = fromAccountOption?.accountType?.id
+        val fromAccountId = fromAccountOption?.accountId
+        val toOfficeId = toAccountOption?.officeId
+        val toClientId = toAccountOption?.clientId
+        val toAccountType = toAccountOption?.accountType?.id
+        val toAccountId = toAccountOption?.accountId
+        val transferDate = DateHelper.getSpecificFormat(
+            DateHelper.FORMAT_dd_MMMM_yyyy,
+            getTodayFormatted(),
+        )
+        val transferAmount = binding.tvAmount.text.toString().toDouble()
+        val transferDescription = binding.tvRemark.text.toString()
+        var dateFormat: String = "dd MMMM yyyy"
+        var locale: String = "en"
+        val fromAccountNumber = fromAccountOption?.accountNo
+        val toAccountNumber = toAccountOption?.accountNo
         if (transferType == TransferType.SELF) {
-            presenter?.makeSavingsTransfer(payload)
+            viewModel.makeSavingsTransfer(
+                fromOfficeId,
```

Comment:
Do we need to create a variable, what about if we pass the value directly like this:

```
viewModel.makeSavingsTransfer(
     fromAccountOption?.officeId,
    ...
}
```

Reply:
done

Reply:
Are you sure? :)

Reply:
my bad , forgot to push the code :(